include/glow/Runtime/Provisioner/Provisioner.h

```diff
@@ -64,10 +64,28 @@ class Provisioner final {
   /// List of available DeviceManagers added during initialization.
   std::vector<DeviceManager *> devices_;
 
-  /// Helper function to cleanup a provision call. On a success free resources
-  /// that are no longer needed by the compiledFunctions. On failure free the
+  /// Helper function to cleanup a provision call. On failure free the
   /// compiledFunctions that were created.
   void cleanupProvision(llvm::ArrayRef<std::string> names, bool failure = true);
+
+  /// Helper function to parse the DAG and generate logicalDevices.
+  std::map<DeviceIDTy, std::vector<DAGNode *>>
+  generateLogicalDevices(DAGListTy &networks);
```

Comment:
Function argument could be `const &`, right?

Also please check all other functions you introduced and make arguments `const` where applicable.